tracer/src/Datadog.Trace/IAST/IastRequestContext.cs

```diff
@@ -41,6 +44,33 @@ internal void AddVulnerability(Vulnerability vulnerability)
         }
     }
 
+    internal void AddRequestBody(object body)
+    {
+        try
+        {
+            var keysAndValuesDic = ObjectExtractor.Extract(body) as Dictionary<string, object>;
+
+            if (keysAndValuesDic != null)
+            {
+                foreach (var key in keysAndValuesDic.Keys)
+                {
+                    var value = keysAndValuesDic[key] as string;
+
+                    if (!string.IsNullOrEmpty(value))
+                    {
+#pragma warning disable CS8604 // Possible null reference argument.
+                        _taintedObjects.TaintInputString(value, new Source(SourceType.GetByte(SourceTypeName.RequestBody), key, value));
```

Comment:
The `key` is also an untrusted source that should be tainted, for example:

```json
{ 
  "filter" : {
    "username": "john"
  }
}
```
can end up in a query like (`_` marks the tainted ranges):

```sql
select * from users where _username_ = '_john_'
```

Reply:
The key of a dictionary inside the json can be used, yes. The updated code takes that into account. Thanks!